### PR TITLE
[BEAM-22] Use an explicit coder in XmlSourceTest

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace "<Jira issue #>" in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

The standard coder for an XML source is the JAXBCoder, which is not
suitable as part of a composite coder.

This is required for [[BEAM-22]](https://issues.apache.org/jira/browse/BEAM-22), due to the implementation of
CreatePCollectionView.

### CODE REVIEW

**Comment by tgroh** [comment]
R: @bjchambers

**Comment by bjchambers** [comment]
@dhalperi for discussion. If `XmlCoder` is not suitable for intermediate usage, should we really have `XmlSource` set that as the coder for its output?

**Comment by bjchambers** [comment]
LGTM, will merge pending discussion.

File: sdks/java/core/src/test/java/com/google/cloud/dataflow/sdk/io/XmlSourceTest.java [hunk 1]
```diff
@@ -665,7 +672,9 @@ public void testReadXMLLargeDataflow() throws IOException {
             .withRecordElement("train")
             .withRecordClass(Train.class)
             .withMinBundleSize(1024);
-    PCollection<Train> output = p.apply(Read.from(source).named("ReadFileData"));
+    AvroCoder<Train> intermediateCoder = AvroCoder.of(Train.class);
+    PCollection<Train> output =
+        p.apply(Read.from(source).named("ReadFileData")).setCoder(intermediateCoder);
```
Comment (dhalperi) [inline sdks/java/core/src/test/java/com/google/cloud/dataflow/sdk/io/XmlSourceTest.java]: please update to the happy `apply` syntax (aka, drop 'named').

**Comment by dhalperi** [comment]
R: @dhalperi so the scripts pick it up.

**Comment by dhalperi** [comment]
Please put `[BEAM-22]` in the PR/commit description?

**Comment by dhalperi** [comment]
My main comment is that we should probably fix JAXBCoder to behave correctly in inner context. That is backwards-incompatible, but it fixes a bug that is probably causing crashes if users are using it this way.

**Comment by dhalperi** [comment]
R: @davorbonaci to comment on backwards-incompatible change.

**Comment by tgroh** [comment]
Replaced by #116

**Pull request closed without merging**